pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTModuleUsesDirective.java

```diff
@@ -0,0 +1,26 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.java.ast;
+
+/**
+ * A "uses" directive of a {@linkplain ASTModuleDeclaration module declaration}.
+ *
+ * <pre class="grammar">
+ *
+ * ModuleUsesDirective ::= "uses" &lt;PACKAGE_NAME&gt; ";"
+ *
+ * </pre>
+ */
+public final class ASTModuleUsesDirective extends AbstractPackageNameModuleDirective {
```

Comment:
```suggestion
public final class ASTModuleUsesDirective extends ASTModuleDirective {
```

Probably a helper method like in provides makes sense:

```java
    public ASTClassOrInterfaceType getService() {
        return firstChild(ASTClassOrInterfaceType.class);
    }
```